classes/Event/class.xoctEventGUI.php

```diff
@@ -1427,30 +1436,41 @@ protected function createHyperlinks(string $intro_text): string
         return $intro_text;
     }
 
-    protected function addCurrentUserToProducers(): void
+    /**
+     * Adds the current user to the specified group.
+     * Producers group is the default selected group. (PluginConfig::F_GROUP_PRODUCERS)
+     *
+     * @param string $group_config_name the group config name (default PluginConfig::F_GROUP_PRODUCERS)
+     *
+     */
+    protected function addCurrentUserToGroup(string $group_config_name = PluginConfig::F_GROUP_PRODUCERS): void
     {
         $xoctUser = xoctUser::getInstance($this->user);
-        // add user to ilias producers
+        // add user to the group
         $sleep = false;
         try {
-            if ($group_producers = PluginConfig::getConfig(PluginConfig::F_GROUP_PRODUCERS)) {
-                $ilias_producers = Group::find($group_producers);
-                $sleep = $ilias_producers->addMember($xoctUser);
+            $group_config_value = PluginConfig::getConfig($group_config_name);
+            if (!empty($group_config_value)) {
+                $group_obj = Group::find($group_config_value);
```

Comment:
Is there a `Group` in any case with any `$group_config_value` passed? maybe catch that for sanity and throw an exception otherwise

Reply:
Because the group configs are optional, meaning it could be empty for some institutions, therefore, I would not suggest throwing error, but to maintain the way it is to skip the procedure if empty!

Reply:
I meant rather, if `$group_config_value` is set, but `Group::find($group_config_value);` then returns null, because this group does not exist. Is that possible?

Reply:
The `$group_config_value` can hold the actual OC group name set by the user (it can be empty)
the `$group_config_name` would always have a value because it is something that needs to be implemented by a developer!

`Group::find($group_config_value);` you mentioned is not correct and has to be `Group::find($group_config_name);`

So, the sanity check does not play any sensible role.

Reply:
By looking into the code of `Group` and `APObject` class, I would say, the `Group::find(...)` only works as a cache lookup and it will never return null but an api instance. 
What could go wrong is by calling `addMember` function and when the group does not exists in the Opencast. However, this behavior has never been handled throught the code, e.g.
- https://github.com/opencast-ilias/OpenCast/blob/df11ddbdc334a01cef332925ff58841644f2636c/classes/class.ilObjOpenCastGUI.php#L390
- https://github.com/opencast-ilias/OpenCast/blob/df11ddbdc334a01cef332925ff58841644f2636c/classes/Service/class.xoctSeriesAPI.php#L151